components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -123,31 +129,39 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 		return errors.Wrap(err, "ProcessIncomingNode unable to marshal projects data")
 	}
 
+	var mgrType string
+	if node.ManagerId != "" {
+		mgrType, err = db.SelectStr(sqlGetManagerTypeFromId, node.ManagerId)
+		if err != nil {
+			logrus.Warnf("unable to find manager for node")
+		}
+	}
```

Comment:
It'd probably be more efficient to do this with a join, but modifying the 4 insert queries does not sound like fun. :(

Reply:
agreeeed